src/input/input_python.rs

```diff
@@ -381,6 +381,28 @@ impl<'a> Input<'a> for PyAny {
             err_val_error!(input_value = self.as_error_value(), kind = ErrorKind::TupleType)
         }
     }
+
+    fn strict_timedelta(&self) -> ValResult<EitherTimedelta> {
+        if let Ok(dt) = self.cast_as::<PyDelta>() {
+            Ok(dt.into())
+        } else {
+            err_val_error!(input_value = self.as_error_value(), kind = ErrorKind::TimedeltaType)
+        }
+    }
+
+    fn lax_timedelta(&self) -> ValResult<EitherTimedelta> {
+        if let Ok(dt) = self.cast_as::<PyDelta>() {
+            Ok(dt.into())
+        } else if let Ok(str) = self.extract::<String>() {
```

Comment:
I think it should be marginally faster to do `cast_as::<PyString>()... py_str.to_string_lossy().to_string().into_bytes()`, particularly for the case where it's not a string.